.github/scripts/collect-metrics.sh

```
@@ -0,0 +1,225 @@
+
+# Function to calculate duration in minutes
+calculate_duration() {
+    local start="$1"
+    local end="$2"
+    
+    if [[ -z "$start" || -z "$end" ]]; then
+        echo "0"
+        return
+    fi
+    
+    local start_epoch=$(date -j -f "%Y-%m-%dT%H:%M:%SZ" "$start" "+%s" 2>/dev/null || echo "0")
```

Shellcheck (0.11.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)